### DISCUSSION TITLE
App detects LSPosed

### DISCUSSION THREAD

**Discussion by clarenze666** [LSPosed/LSPosed#2715] [Q&A]
Hi all,

I have an app that will crash as soon as I activate a module in LSPosed for this app (any module, I tried many). In logcat I see following message:

`Abort message: 'No pending exception expected: java.lang.ClassNotFoundException: com.medtronic.minimed.data.utilities.MobileSecurityUtil
`
It seems it somehow detects LSPosed and hides some of its classes. I tried to run a decompiler on the app, but the apk basically only consists of .so files.
Any ideas how I can get around this?

I am running LineageOs 20 (Android 13) on a Oneplus 7pro. Magisk and LSPosed are on the latest released version.

Regards

Clarenze

**Comment by infinity0** [comment]
See also #2055 and #2686, unfortunately devs don't care.

**Comment by PlatFlat** [comment]
I tested 2 apps. Revolut v9.22 & DBS PayLah v5.9.4, when any LSPosed module is hooked to that two apps. Instant crash when launch.
I think it will be a major issue in coming months, many apps will adopt the same detection methods.

**Comment by infinity0** [comment]
My current workaround is to put those apps in a new user profile, at least then they won't be able to see anything else on the system.

In case they do root detection, you also have to install them on the main user profile (using `pm install-existing --user 0`) in order to add them to the Magisk denylist in order for Shamiko to hide it. topjohnwu/Magisk#7232
